tests/ui/char_slices.rs

```diff
@@ -0,0 +1,15 @@
+#![warn(clippy::char_slices)]
```

Comment:
This is missing the corresponding `char_slices.stderr` file. If you run the tests, you should get a message telling you how to "bless" the result of the test, and then you can commit the file. This will likely also make CI pass.

Reply:
This is because it currently does not produce any output in stderr. I think something is wrong but I have no idea what it is. If you have any ideas they would be very welcome. (This is only my second PR to clippy, and my first was relatively easy because it was almost exactly the same as three other lints).

Reply:
Hmmm... What I usually do in this case is to comment out lint conditions or move the lint out of other checks until output appears. Then I may reorder the checks until I've found the one that precludes linting and then I try to understand why it doesn't match.

Reply:
@llogiq I think I found the problem. `res.opt_def_id()` is never equal to `cx.tcx.lang_items().char_impl()`. What's another or better way to see if the inner type of a slice is char?

Reply:
IMHO How about just using `rustc_middle::ty::Char` to check a type in slice? I guess the `hir_ty_to_ty` method can convert from `rustc_hir::Ty` to `ty::Ty`.